experimental/uefi/runtime/src/remote_attestation.rs

```diff
@@ -22,12 +22,31 @@
 extern crate alloc;
 
 use alloc::vec::Vec;
-use anyhow::Context;
-use oak_remote_attestation_sessions::{SessionId, SessionState, SessionTracker};
+use anyhow::{bail, Context};
+use oak_remote_attestation_sessions::{SessionId, SessionState, SessionTracker, SESSION_ID_LENGTH};
 
 /// Number of sessions that will be kept in memory.
 const SESSIONS_CACHE_SIZE: usize = 10000;
 
+fn deserialize_request(msg: Vec<u8>) -> anyhow::Result<(SessionId, Vec<u8>)> {
+    let mut session_id: SessionId = [0; SESSION_ID_LENGTH];
+    let mut request_body: Vec<u8> = Vec::with_capacity(msg.len() - SESSION_ID_LENGTH);
+
+    if msg.len() < SESSION_ID_LENGTH {
+        bail!("Message too short to contain a SessionId. The length of a SessionId is {} bytes, the message received contained only {} bytes", SESSION_ID_LENGTH, msg.len());
+    }
+
+    for (index, byte) in msg.into_iter().enumerate() {
```

Comment:
nit: I think that splitting the slice (https://doc.rust-lang.org/std/primitive.slice.html#method.split_at) and then using `copy_to_slice` for the message id and `extend_from_slice` for the request body might make this more readable.